packages/core/src/modules/vc/data-integrity/W3cJsonLdCredentialService.ts

```diff
@@ -265,6 +304,52 @@ export class W3cJsonLdCredentialService {
         challenge: options.challenge,
         domain: options.domain,
         documentLoader: this.w3cCredentialsModuleConfig.documentLoader(agentContext),
+        checkStatus: async ({ credential }: { credential: W3cJsonCredential }) => {
+          // Only throw error if credentialStatus is present
+          if ('credentialStatus' in credential) {
+            if (Array.isArray(credential.credentialStatus)) {
+              throw new CredoError(
+                'Verifying credential status as an array for JSON-LD credentials is currently not supported'
+              )
+            }
+
+            // Ensure credentialStatus contains the necessary properties
+            if (!credential.credentialStatus || credential.credentialStatus.statusListIndex === undefined) {
+              throw new CredoError('Invalid credential status format')
+            }
+
+            const credentialStatusURL = credential.credentialStatus.statusListCredential
```

Comment:
We should first check the type fiets to make sure this is a bit string status type. There could be others and those are not supported at the moment.